### PR TITLE
feat: raw bytecode analysing script

### PR DESCRIPTION
**Pull request by arianejasuwienas**
**Description**:
Raw bytecode analysing tool

### CODE REVIEW

File: tools/custom/raw-bytecode-analyser/README.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
+# Hedera Smart Contract Bytecode Analyser
+
+This Python script analyses the bytecode of Smart Contracts on the Hedera network to detect specific operand usages.
+It can help determine the likelihood that a Smart Contract creates a token using an SECP256k1 key.
```
Comment (Nana-EC) [inline tools/custom/raw-bytecode-analyser/README.md:4]:
```suggestion
This following example determines the likelihood that a Smart Contract creates a token using an SECP256k1 key.
```

File: tools/custom/raw-bytecode-analyser/README.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+## Usage
+
+The script can be run from the command line with the following arguments:
+
+- `contract_id`: The ID of the smart contract whose bytecode you want to analyze.
+- `--previewnet`: Optional flag to use the Previewnet Mirror Node URL.
+- `--testnet`: Optional flag to use the Testnet Mirror Node URL.
+
+If no network flag is specified, the script defaults to using the Mainnet Mirror Node URL.
```
Comment (Nana-EC) [inline tools/custom/raw-bytecode-analyser/README.md:27]: I would prefer we switch this. Default to testnet and let people set a flag to point to mainnet.

File: tools/custom/raw-bytecode-analyser/README.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+```bash
+pip install requests requests-cache evmdasm
+```
+
+## Usage
+
+The script can be run from the command line with the following arguments:
+
+- `contract_id`: The ID of the smart contract whose bytecode you want to analyze.
```
Comment (Nana-EC) [inline tools/custom/raw-bytecode-analyser/README.md:23]: Please support contract_id or contract evm address. This way devs from either world can get going quick

**Review by Nana-EC — COMMENTED** [review]
Looking good.
Left some comments.

**Review by acuarica — DISMISSED** [review]
lgtm

File: tools/custom/raw-bytecode-analyser/detect_token_creation_with_secp_key.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+            evm_bytecode = EvmBytecode(bytecode)
+            disassembled_code = evm_bytecode.disassemble()
+            for instruction in disassembled_code:
+                if instruction.name[:4] != 'PUSH':
+                    continue
+                if instruction.operand == '0167':
+                    address_detected = True
+                if instruction.operand in ['0fb65bf3', '2af0c59a', 'ea83f293', 'abb54eb5']:
+                    method_detected = True
+                if instruction.operand == '03' and instruction.name == 'PUSH1':
```
Comment (david-bakin-sl) [inline tools/custom/raw-bytecode-analyser/detect_token_creation_with_secp_key.py:27]: Might be worth (for self-documenting purposes) to give a name to `'03'`.

File: tools/custom/raw-bytecode-analyser/detect_token_creation_with_secp_key.py [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    if response.status_code == 200:
+        bytecode = response.json().get('runtime_bytecode')
+        if bytecode:
+            address_detected = False
+            method_detected = False
+            param_detected = False
+            call_detected = False
+
+            evm_bytecode = EvmBytecode(bytecode)
+            disassembled_code = evm_bytecode.disassemble()
```
Comment (david-bakin-sl) [inline tools/custom/raw-bytecode-analyser/detect_token_creation_with_secp_key.py:19]: Amazing how this library makes this simple script so easy!

**Review by david-bakin-sl — COMMENTED** [review]
Possible bug?

**Comment by david-bakin-sl** [comment]
P.S. I like the way the readme carefully documents the limitations.  This is a great script to easily / quickly find the contracts that might qualify, then you need to investigate more closely to really understand them.  As people adapt this script for their own investigations they'll need to keep this in mind ...

**Review by acuarica — APPROVED** [review]
lgtm

**Pull request merged**